config/tools/generator/main.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"go/format"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+	"github.com/scaleway/provider-scaleway/config/tools"
+)
+
+// ProviderGoFilePath is the path to the provider.go file where provider configurations are set.
+const ProviderGoFilePath = "config/provider.go"
+
+// ExternalNameGoFilePath is the path to the external_name.go file where external name configurations are defined.
+const ExternalNameGoFilePath = "config/external_name.go"
+
+func parseTemplates() (*template.Template, error) {
+	templatePath := "config/config_templates.tmpl"
```

Comment:
Could it be a embedding static file? That way if we try to compile if the file is missing/moved the compilation step won't pass.